### PR TITLE
Rewrite build system using webpack

### PR DESCRIPTION
**Pull request by saharj**
This is a large pull request.

**Motivation**
Grunt as a built tool is becoming legacy software. It's not well maintained. Because of Grunt and Grunt specific packages that require native modules down in their dependency tree building in Windows has always been a pain. Using webpack should modernize the build system and also make it possible to build on Windows without compiling any native modules. In this pull request I removed many legacy dependencies and implemented a better and cleaner build system using webpack and npm scripts. 

All internal and external dependencies are now using `require` statements. Even CSS code is loaded via `require`. This pull request makes it easier to use new ES6 language features in future and also avoids bundling dead code. 

**This pull request removes:**
* [x] Dependency to Bower
* [x] Dependency to JSCS
* [x] Dependency to JSHint
* [x] Committed to source code fonts in `/app/fonts` folder 
* [x] Grunt build system
* [x] Dependency to worker wrapper modules (yaml-worker and sway-worker)
* [x] Two packages system (`swagger-editor-src` and `swagger-editor`). `swagger-editor-src` will get deprecated. 

**This pull request adds**
* [x] webpack build system
* [x] ESLint for code linting
* [x] New npm scripts for test
* [x] webpack based unit test dependency resolution
* [x] A single package (`swagger-editor`) in npm. The published package can be used as a static set of files yet you can `npm install` and `npm start` to run in development mode. As before no server will come with the package. 

**TODO**

* [x] `npm run unit-test-watch` should work 
* [x] ESLint should pass. We are using Google defaults and there are many discrepancies. 
* [x] End to End tests configuration 
* [x] All end to End tests should pass
* [x] re-include unit tests in `index.js` at `/test/unit` and make them all pass
* [x] Update documentations accordingly
* [x] Update Docker build config if necessary  
* [x] Fix TravisCI build
* [x] Open browser when `npm start` finished. (Only in development mode)

This should go live as part of 2.10.x release

### CODE REVIEW

File: scripts/controllers/editor.js [hunk 1]
```diff
@@ -14,6 +14,10 @@ SwaggerEditor.controller('EditorCtrl', function EditorCtrl($scope, $rootScope,
     }
   });
 
+/**
+ * Get Debounced On Ace Change.
+ * @constructor
```
Comment (saharj) [inline scripts/controllers/editor.js]: This not a constructor. Specify the return type value (function returns debounced unchanged function)

File: scripts/controllers/editor.js [hunk 2]
```diff
@@ -32,6 +36,7 @@ SwaggerEditor.controller('EditorCtrl', function EditorCtrl($scope, $rootScope,
     });
   });
 
+/** When there is a change on ace */
```
Comment (saharj) [inline scripts/controllers/editor.js]: Indentation and multiline

**Comment by saharj** [comment]
Works flawlessly on Windows:  🎉 

screen shot 2016-04-11 at 9 09 27 pm

**Comment by fehguy** [comment]
awesome!

**Pull request merged**